modules/core/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManager.java

```diff
@@ -28,9 +35,29 @@
      * @param objName Object to collect statistics by.
      * @param colNames Columns to collect statistics by.
      * @throws IgniteCheckedException  Throws in case of errors.
+     * @return future to get fini
      */
     public void collectObjectStatistics(String schemaName, String objName, String... colNames) throws IgniteCheckedException;
 
+    /**
+     * Collect objects statistics.
+     *
+     * @param keys Collection of keys to collect statistics by (schema, obj, columns).
+     * @return Future to track progress and cancel collection.
+     * @throws IgniteCheckedException In case of errors.
+     */
+    public StatsCollectionFuture<Map<GridTuple3<String, String, String[]>, ObjectStatistics>> collectObjectStatisticsAsync(
```

Comment:
Had to duplicate StatisticsKeyMessage as StatisticsTarget because Message shouldn't be in core interface and messages shouldn't inherit fields. Fixed